test/mainpanel.test.tsx

```diff
@@ -2,31 +2,163 @@ import { mount } from 'enzyme'
+  })
+
+  it('should forward to preview location and back again', async () => {
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo}>
+          <MainPanel {...MainPanelProps} />
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    const contentlist = wrapper.update().find(ContentList)
+    const tablerow = contentlist.find(TableBody).find(TableRow)
+
+    act(() => {
+      ;(tablerow.first().prop('onClick') as any)({ target: { innerHTML: TestContentCollection[0].DisplayName } })
+    })
+
+    expect(window.location.pathname).toEqual(`/preview/${TestContentCollection[0].Id}`)
+
+    const backbtn = wrapper.find(Button)
+    await act(async () => {
+      ;(backbtn.first().prop('onClick') as any)()
+    })
+    const foldernametitle = wrapper.update().find(Typography)
+    expect(foldernametitle.text()).toEqual(`/`)
+  })
+
+  it('should open a folder', async () => {
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo}>
+          <MainPanel {...MainPanelProps} />
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    const contentlist = wrapper.update().find(ContentList)
+    const tablerow = contentlist.find(TableBody).find(TableRow)
+
+    await act(async () => {
+      ;(tablerow.at(1).prop('onDoubleClick') as any)({
+        target: { innerHTML: TestContentCollection[1].DisplayName },
+      })
+    })
+
+    const foldernametitle = wrapper.update().find(Typography)
+    expect(foldernametitle.text()).toEqual(`/${TestContentCollection[1].Name}`)
+  })
+
+  it('should download a content', async () => {
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo}>
+          <MainPanel {...MainPanelProps} />
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    const downloadicon = wrapper.update().find(CloudDownload)
+    act(() => {
+      ;(downloadicon.prop('onClick') as any)()
+    })
+
+    expect(downloadFile).toBeCalled()
+  })
+
+  it('should open edit mode', async () => {
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo}>
+          <MainPanel {...MainPanelProps} />
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    const editbtn = wrapper.update().find(Edit)
+    act(() => {
+      ;(editbtn.first().prop('onClick') as any)({ content: { Id: TestContentCollection[0].Id } })
+    })
+
+    expect(window.location.pathname).toEqual(`/edit/${TestContentCollection[0].Id}`)
+  })
+
+  it('should order the list', async () => {
+    await act(async () => {
+      wrapper = mount(
+        <RepositoryContext.Provider value={repo}>
+          <MainPanel {...MainPanelProps} />
+        </RepositoryContext.Provider>,
+      )
+    })
+
+    const orderbtn = wrapper.update().find(TableSortLabel)
+    act(() => {
+      ;(orderbtn.first().prop('onClick') as any)()
+    })
```

Comment:
How do you know it is ordered after the click?

Reply:
Yes, I fixed